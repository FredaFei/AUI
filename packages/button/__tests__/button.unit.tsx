import * as renderer from 'react-test-renderer'
import React from 'react'
import {configure, render, shallow} from 'enzyme'
import Adapter from 'enzyme-adapter-react-16';

configure({
  adapter: new Adapter(),
});
import Button from '../button'

describe('Button', () => {
  it('Button 存在', () => {
    const json = renderer.create(<Button>hello</Button>).toJSON()
    expect(json).toMatchSnapshot()
  })
  it('Button 带 icon,iconPosition默认left', () => {
    const wrapper = render(<Button icon="settings">设置</Button>)
    console.log(wrapper)
    // expect(wrapper.props().icon).toBe('settings')
    // expect(wrapper.props().iconPosition).toBe('left')
  })
  it('Button 带 icon,iconPosition可设置', () => {
    const wrapper = shallow(<Button iconPosition="right">设置</Button>)
    console.log(wrapper)
    expect(wrapper.props().iconPosition).toBe('right')
  })
  it('Button 设置 onClick', () => {
    const mockCallBack = jest.fn();
    const wrapper = shallow(<Button onClick={mockCallBack}>Ok!</Button>);
    wrapper.simulate('click');
    expect(mockCallBack.mock.calls.length).toEqual(1);
  });
  it('Button 设置 自定义类名', () => {
    const wrapper = shallow(<Button className="test">设置</Button>)
    expect(wrapper.find('button').hasClass('test')).toBe(true)
  })
  it('Button 设置 自定义样式', () => {
    const wrapper = shallow(<Button style={{color: 'red'}}>设置</Button>)
    expect(wrapper.render()).toMatchSnapshot()
  });
})