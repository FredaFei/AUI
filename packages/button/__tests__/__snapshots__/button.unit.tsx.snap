// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试 Button 以下行为 Button 存在 1`] = `
<button
  className="am-button-wrapper "
  data-role="Button"
  onClick={[Function]}
>
  <span
    className="am-button-content"
  >
    hello
  </span>
</button>
`;

exports[`测试 Button 以下行为 Button 设置 自定义样式 1`] = `
initialize {
  "0": Object {
    "attribs": Object {
      "class": "am-button-wrapper ",
      "data-role": "Button",
      "style": "color:red",
    },
    "children": Array [
      Object {
        "attribs": Object {
          "class": "am-button-content",
        },
        "children": Array [
          Object {
            "data": "设置",
            "next": null,
            "parent": [Circular],
            "prev": null,
            "type": "text",
          },
        ],
        "name": "span",
        "namespace": "http://www.w3.org/1999/xhtml",
        "next": null,
        "parent": [Circular],
        "prev": null,
        "type": "tag",
        "x-attribsNamespace": Object {
          "class": undefined,
        },
        "x-attribsPrefix": Object {
          "class": undefined,
        },
      },
    ],
    "name": "button",
    "namespace": "http://www.w3.org/1999/xhtml",
    "next": null,
    "parent": null,
    "prev": null,
    "root": Object {
      "attribs": Object {},
      "children": Array [
        [Circular],
      ],
      "name": "root",
      "namespace": "http://www.w3.org/1999/xhtml",
      "next": null,
      "parent": null,
      "prev": null,
      "type": "root",
      "x-attribsNamespace": Object {},
      "x-attribsPrefix": Object {},
    },
    "type": "tag",
    "x-attribsNamespace": Object {
      "class": undefined,
      "data-role": undefined,
      "style": undefined,
    },
    "x-attribsPrefix": Object {
      "class": undefined,
      "data-role": undefined,
      "style": undefined,
    },
  },
  "_root": [Circular],
  "length": 1,
  "options": Object {
    "decodeEntities": true,
    "normalizeWhitespace": false,
    "withDomLvl1": true,
    "xml": false,
  },
}
`;
